pyproject.toml

```diff
@@ -110,8 +111,58 @@ disable = [
     "fixme",
 ]
 
-[tool.pylint-nautobot]
-supported_nautobot_versions = ["1", "2"]
+[tool.ruff]
+line-length = 100
+target-version = "py310"
+
+[tool.ruff.lint]
+select = [
+    "B",  # flake8-bugbear
+    "D",  # pydocstyle
+    "E",  # pycodestyle error
+    "F",  # pyflakes
+    "I",  # isort
+    "PL", # pylint
+    "S",  # bandit
+    "W",  # pycodestyle warning
+]
+ignore = [
+    # D105 enforces docstrings for magic methods, e.g. __str__(). It seems unnecessary.
+    "D105",
+
+    # warning: `one-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible.
+    "D203", # 1 blank line required before class docstring
+
+    # D212 is enabled by default in google convention, and complains if we have a docstring like:
+    # """
+    # My docstring is on the line after the opening quotes instead of on the same line as them.
+    # """
+    # We've discussed and concluded that we consider this to be a valid style choice.
+    "D212", # Multi-line docstring summary should start at the first line
+]
+
+[tool.ruff.lint.isort]
+force-sort-within-sections = true
+
+[tool.ruff.lint.pycodestyle]
+# Combine line-length and max-line-length to emulate flake8-bugbear's B950 rule.
+# The formatter uses line-length, but rule E501 won't complain until a line hits 111.
+max-line-length = 110
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
+
+[tool.ruff.lint.per-file-ignores]
+"nautobot_fsus/**/__init__.py" = [
+    "D",
```

Comment:
not familiar with what these "D" and "S" are doing?

Reply:
"D" is pydocstyle, and "S" is flake8-bandit. Handy list of ruff rules and their codes: https://docs.astral.sh/ruff/rules/